### PR TITLE
New clara dev

### PR DESCRIPTION
**Pull request by benjgrad**

### CODE REVIEW

File: exception_handler.c [hunk 1]
```diff
@@ -92,26 +94,31 @@ void the_exception (void)
...
 
+  asm ("movia r22, process_stack_pointer)");
+  asm ("stw sp, 0(r22)");
+
   //move in address of ksp var into register
-  asm("movia r22, ksp"); //compiles if i dont do &ksp, but ksp instead
+  asm ("movia r23, ksp"); //compiles if i dont do &ksp, but ksp instead
   //store sp register value into kernel stack address, which needs to be in a register
-  asm("stw sp, 0(r22)");
+  asm ("ldw sp, 0(r23)");
```
Comment (usernameisntallowed) [inline exception_handler.c]: I think this line takes the 1st value at the address specified by the kernel stack pointer and moves it into sp which is not what we want.

File: qcore.c [hunk 2]
```diff
@@ -11,132 +11,89 @@ DESCRIPTION:	The core of the QUERK kernel. Includes the software interrupt
...
-static  char    kernel_stack[512];
-static  void    *kernel_stack_pointer = (void *) &kernel_stack[511];
-// Ask why we can't just define this in the exception handler and
-
-static  int     num_of_processes = 0;	/* counter used to assign unique pids */
-
-static  Process     process_array[MAX_NUM_OF_PROCESSES]; // Formerly pdb_array
 extern unsigned int process_stack_pointer;
-extern unsigned int ksp;
+extern unsigned int ksp = &kernel_stack_pointer;
```
Comment (jklasdfj) [inline qcore.c]: Going to investigate and test removing the unnecessary 2nd level of indirection as instructed by supervisor.

File: qcore.c [hunk 3]
```diff
@@ -11,132 +11,89 @@ DESCRIPTION:	The core of the QUERK kernel. Includes the software interrupt
...
 {
-	int control;
-	control = *(JTAG_UART_ptr + 1);	// read the JTAG_UART Control register
+  int control;
+  control = *(JTAG_UART_ptr + 1); // read the JTAG_UART Control register
 
-	if(control & 0xFFFF0000) {// if space, then echo character, else ignore
-		*(JTAG_UART_ptr) = c;
-	}
+  if(control & 0xFFFF0000) { // if space, then echo character, else ignore
```
Comment (jklasdfj) [inline qcore.c]: Self-reminder, update to do proper polling as demonstrated.

File: qcore.c [hunk 4]
```diff
@@ -11,132 +11,89 @@ DESCRIPTION:	The core of the QUERK kernel. Includes the software interrupt
...
-        //create new process
-        new_process		= &process_array[new_pid];	/* pointer to descriptor */
-        new_process->pid	= new_pid;
-        new_process->state	= Ready;
-        new_process->user_stack_pointer		= stack_bottom + stack_size; // XXX: Confirm that this is correct
-
-        AddToTail (&ready_queue, new_process);
+  asm ("movia r12, temp_sp");
+  asm ("ldw r13, 0(r12)");
+  asm ("stw r26, 104(r13)");
```
Comment (jklasdfj) [inline qcore.c]: self reminder gp alias

File: request.s [hunk 5]
```diff
@@ -19,7 +19,9 @@ KernelRelinquish:
     subi sp, sp, 4
     stw r5, 0(sp)
     movi r5,1 #relinquish enum
+    addi sp, sp, 4
```
Comment (usernameisntallowed) [inline request.s]: self reminder investigate this

**Review by jharon — APPROVED** [review]

**Pull request merged**